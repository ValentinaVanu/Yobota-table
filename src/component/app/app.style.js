import styled from 'styled-components'

export const StyledAppWrapper = styled.div`
  padding: 2rem 4rem;
  height: 100%;

  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(4, 1fr);
`
