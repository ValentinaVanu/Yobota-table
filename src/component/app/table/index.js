export { YobotaTable } from'./table.component'
