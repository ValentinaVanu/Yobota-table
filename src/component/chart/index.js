export { YobotaChart } from './chart.component'

