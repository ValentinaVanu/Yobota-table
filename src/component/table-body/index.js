export { YobotaTableBody } from './table-body.component'
