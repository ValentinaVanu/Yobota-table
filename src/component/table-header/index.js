export { YobotaTableHeader } from'./table-header.component'
