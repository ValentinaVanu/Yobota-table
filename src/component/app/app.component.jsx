import React from 'react'
import { Provider } from 'react-redux'
import store from '../../store'
import { YobotaChart } from '../chart'
import { YobotaTableHeader } from '../table-header'
import { StyledAppWrapper } from './app.style'

const App = () => {
  return (
    <StyledAppWrapper>
      <Provider store={store}>
        <YobotaTableHeader />
        <YobotaChart />
      </Provider>
    </StyledAppWrapper>
  )
}

export { App }
