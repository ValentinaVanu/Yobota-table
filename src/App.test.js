import {App} from './component/app'
import renderer from 'react-test-renderer'

test('renders main app component', () => {
  const app = renderer.create(<App />).toJSON()
  expect(app).toMatchSnapshot()
})
