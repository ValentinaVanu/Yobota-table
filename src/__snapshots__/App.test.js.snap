// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main app component 1`] = `
<div
  className="sc-gKsewC kCeqVS"
>
  <div
    className="sc-jSgupP bgzEf"
  >
    <div
      className="MuiPaper-root makeStyles-root-2 MuiPaper-elevation6 MuiPaper-rounded"
    >
      <div
        className="MuiTableContainer-root makeStyles-container-3"
      >
        <table
          aria-label="sticky table"
          className="MuiTable-root MuiTable-stickyHeader"
          role={null}
        >
          <thead
            className="MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root sc-hKgILt cbDwcG MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head sc-eCssSg KulhZ MuiTableCell-alignLeft MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 100,
                  }
                }
              >
                <div
                  onClick={[Function]}
                >
                  Last Name
                </div>
                <input
                  id="last_name"
                  name="last_name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                  variant="outlined"
                />
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head sc-eCssSg KulhZ MuiTableCell-alignLeft MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 100,
                  }
                }
              >
                Email
                <br />
                <input
                  id="email"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                  variant="outlined"
                />
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head sc-eCssSg KulhZ MuiTableCell-alignLeft MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 170,
                  }
                }
              >
                DOB
                <svg
                  aria-hidden={true}
                  chart="off"
                  className="MuiSvgIcon-root sc-gsTCUz kWySRt"
                  focusable="false"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 2v20c-5.07-.5-9-4.79-9-10s3.93-9.5 9-10zm2.03 0v8.99H22c-.47-4.74-4.24-8.52-8.97-8.99zm0 11.01V22c4.74-.47 8.5-4.25 8.97-8.99h-8.97z"
                  />
                </svg>
                <br />
                <input
                  id="date_of_birth"
                  name="date_of_birth"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                  variant="outlined"
                />
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head sc-eCssSg KulhZ MuiTableCell-alignLeft MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 100,
                  }
                }
              >
                Industry
                <br />
                <input
                  id="industry"
                  name="industry"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                  variant="outlined"
                />
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head sc-eCssSg KulhZ MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 100,
                  }
                }
              >
                Salary
                <br />
                <input
                  id="salary"
                  name="salary"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                  variant="outlined"
                />
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head sc-eCssSg KulhZ MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 100,
                  }
                }
              >
                Years of Experience
                <br />
                <input
                  id="years_of_experience"
                  name="years_of_experience"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                  variant="outlined"
                />
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <div
        className="MuiTablePagination-root sc-dlfnbm dWjUhw"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            className="MuiTablePagination-spacer"
          />
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            id="mui-96235"
          >
            Rows per page:
          </p>
          <div
            className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-96235 mui-75057"
              className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              id="mui-75057"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              10
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              tabIndex={-1}
              value={10}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            0-0 of 0
          </p>
          <div
            className="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Previous page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Next page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bdfBwQ Dptgt"
  />
</div>
`;
